documentation/SLACK/SLACK.md

```diff
@@ -0,0 +1,764 @@
+   - Severity and status
+   - Description
+   - Creation timestamp
+   - Direct link to the case in AWS console
+5. Case watchers are automatically added to the Slack channel (if they exist in the workspace)
+6. Channel topic is set with case status and severity for quick reference
+
+### 2. Security IR Comments → Slack Messages
+
+**Scenario**: When a comment is added to an AWS Security IR case
+
+**Workflow**:
+1. A comment is added to the case in AWS Security Incident Response
+2. The Security IR Poller Lambda detects the new comment within 1 minute
+3. The comment is automatically posted as a message in the corresponding Slack channel
+4. The message includes author name, content, timestamp, and `[AWS Security IR Update]` tag
+
+### 3. Security IR Watchers/Permissions → Slack Channel Members
+
+**Scenario**: When new watchers or permissions are added to an AWS Security IR case
+
+**Workflow**:
+1. A new watcher is added to the case in AWS Security Incident Response
+2. The Security IR Poller Lambda detects the permission change
+3. The system attempts to find the user in the Slack workspace by email address
+4. **If user exists in Slack workspace**: User is automatically added to the Slack channel with notification
+5. **If user does not exist in Slack workspace**: A notification alert is posted in the channel with instructions for manual invitation
+
+### 4. Slack Messages → Security IR Comments
+
+**Scenario**: When a message is posted in a Slack incident channel
+
+**Workflow**:
+1. A user posts a message in the incident Slack channel
+2. Slack sends a webhook event to the API Gateway endpoint
+3. The Slack Events Bolt Handler Lambda processes the message
+4. The message is automatically added as a comment to the corresponding AWS Security IR case
+5. The comment includes `[Slack Update]` tag, author's real name, message content, and timestamp
+
+### 5. Slack Attachments → Security IR Attachments
+
+**Scenario**: When files are uploaded to a Slack incident channel
+
+**Workflow**:
+1. A user uploads a file to the Slack incident channel
+2. Slack sends a `file_shared` event to the API Gateway endpoint
+3. The Slack Events Bolt Handler Lambda processes the file event
+4. The file is downloaded from Slack using the bot token
+5. The file is uploaded to the corresponding AWS Security IR case as an attachment
+6. File metadata is preserved (filename, size, MIME type)
+7. A confirmation message is posted in Slack: `📎 File uploaded to AWS Security IR case`
+
+**File Handling**:
+- **Size Limit**: Maximum 100MB per file
+- **Supported Formats**: All file types supported by both platforms
+- **Error Handling**: Large files or upload failures generate warning messages
+- **Duplicate Prevention**: Files are checked against existing attachments
+
+### 6. Slash Commands for Case Management
+
+**Available Commands**: The `/security-ir` command supports the following subcommands in incident channels:
+
+#### `/security-ir status`
```

Comment:
issue: We've overloaded the word status here... I can see customers getting confused with this.